Plugins/NativeEngine/Source/VertexBuffer.cpp

```diff
@@ -67,22 +24,31 @@ namespace Babylon
             return;
         }
 
-        m_streams.clear();
-        m_attributes.clear();
-        m_handle.reset();
+        if (bgfx::isValid(m_handle) && m_deviceId == m_deviceContext.GetDeviceId())
+        {
+            if (m_dynamic)
+            {
+                bgfx::destroy(m_dynamicHandle);
+            }
+            else
+            {
+                bgfx::destroy(m_handle);
+            }
+        }
+
         m_bytes.clear();
 
         m_disposed = true;
     }
 
-    void VertexBuffer::Update(const gsl::span<uint8_t> bytes, size_t byteOffset)
+    void VertexBuffer::Update(gsl::span<const uint8_t> bytes, size_t byteOffset)
     {
         if (!m_dynamic)
         {
             throw std::runtime_error{"Cannot update non-dynamic vertex buffer"};
         }
 
-        if (m_handle.has_value())
+        if (bgfx::isValid(m_dynamicHandle))
```

Comment:
Why do we sometimes check m_handle and sometimes m_dynamicHandle? In some of the code it seems like a call to bgfx::isValid with either one is equivalent, and in other places that doesn't seem like the case.

Reply:
It's the same because it is a union. I use m_handle when we don't care which one it is and use m_dynamicHandle if we check it's dynamic. It doesn't actually matter.